### PR TITLE
docs: add utilities docs

### PR DESCRIPTION
**Pull request by mikemurray**
- add a utilities docs with a GraphiQl app setup guide
- add a TOC for the testing doc and utils docs to `docs/readme.md`
- reorganize `docs/readme.md` to move architecture decisions doc to the first position. Seems like it's the most important so it should go first.

### CODE REVIEW

File: docs/utilities.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+## GraphiQL App
+
+1. Follow the instructions to install the [GraphiQL app](https://github.com/skevy/graphiql-app).
+1. Launch the app
+1. Click the button labeled `Edit HTTP Headers`
+1. Click `+ Add Header`
+1. For `Header name` enter `meteor-login-token`
+1. For `Header value` you'll need your `Meteor.loginToken` from the [Getting Started](https://github.com/reactioncommerce/reaction-next-starterkit#getting-started) guide.
+1. TL;DR: With a Reaction App running using docker-compose, open the local storage panel in your browsers dev tools, and copy the value of the `Meteor.loginToken` into the `Header value` field.
```
Comment (machikoyasuda) [inline docs/utilities.md]: Move TL;DR to the top and quote it:

```
> TL;DR: With a Reaction App running using `docker-compose`, open the Local Storage panel in your browser dev tools and copy the value of `Meteor.loginToken` into the `Header value` field
```

Add a last step:
1. Make your first query by setting `GraphQL Endpoint` to `http://localhost:3030/graphql-alpha` and querying for the viewer's name: 

```
{
  viewer {
    name
  }
}
```

This way, people can confirm the installation, header setting and running the apps all work.

For more on your getting started with GraphQL and Reaction: https://docs.reactioncommerce.com/reaction-docs/master/graphql-intro

**Comment by machikoyasuda** [comment]
Looks good 💯

**Comment by mikemurray** [comment]
@machikoyasuda  I do see the docs in the sidebar, maybe you have to restart the app?
